src/network/networkd-ndisc.c

```diff
@@ -214,7 +286,38 @@ static int ndisc_router_process_autonomous_prefix(Link *link, sd_ndisc_router *r
 
         if (in_addr_is_null(AF_INET6, (const union in_addr_union *) &link->network->ipv6_token) == 0)
                 memcpy(((char *)&address->in_addr.in6) + 8, ((char *)&link->network->ipv6_token) + 8, 8);
-        else {
+        else if (link->network->ipv6_address_generation_type == IPV6_TOKEN_ADDRESS_GENERATION_PREFIXSTABLE) {
+                struct in6_addr addr = address->in_addr.in6;
+                IPv6Token *i;
+
+                LIST_FOREACH(tokens, i, link->network->ipv6tokens) {
+                        if (!memcmp(&i->prefix, &addr, FAMILY_ADDRESS_SIZE(address->family))) {
+
+                                for (; i->dad_counter < DAD_CONFLICTS_IDGEN_RETRIES_RFC7217; i->dad_counter++) {
+                                        r = ipv6_makestableprivate(link, &addr, prefixlen, &i->dad_counter, &address->in_addr.in6);
+                                        if (r < 0)
+                                                return log_link_error_errno(link, r, "Could not make stable address: %m");
+
+                                        if (!ipv6_interface_identifiers(&address->in_addr.in6))
+                                                break;
+                                }
+
+                                match = true;
+                        }
+                }
+        } else if (link->network->ipv6_address_generation_type == IPV6_TOKEN_ADDRESS_GENERATION_EUI64) {
+                IPv6Token *i;
+
+                LIST_FOREACH(tokens, i, link->network->ipv6tokens) {
+                        if (!memcmp(&i->token, &address->in_addr, FAMILY_ADDRESS_SIZE(address->family))) {
+                                memcpy(((char *)&address->in_addr.in6) + 8, ((char *)&i->token) + 8, 8);
+                                break;
```

Comment:
Needs to set `match` flag?

Reply:
it will got to `eui64` which is intended.